mon-pix/tests/unit/routes/campaigns/start-or-resume_test.js

```diff
@@ -6,34 +6,108 @@ import sinon from 'sinon';
 describe('Unit | Route | Start-or-resume', function () {
   setupTest();
 
-  let route;
-
-  const campaign = {
-    code: 'NEW_CODE',
-    isRestricted: true,
-  };
+  let route, campaign;
 
   beforeEach(function () {
+    campaign = {
+      code: 'NEW_CODE',
+    };
     route = this.owner.lookup('route:campaigns.start-or-resume');
+    route.modelFor = sinon.stub().returns(campaign);
+    route.campaignStorage = { get: sinon.stub() };
+    route.router.transitionTo = sinon.stub();
   });
 
   describe('#beforeModel', function () {
-    it('should reset state', async function () {
-      // given
-      route.state.campaignCode = 'OLD_CODE';
-      route.state.hasUserSeenJoinPage = true;
-      const transition = { to: { queryParams: {} } };
-      route.modelFor = sinon.stub().returns(campaign);
-      route.session.isAuthenticated = true;
-      route.store = { queryRecord: sinon.stub() };
-      route.currentUser = { user: { mustValidateTermsOfService: false } };
-      route.replaceWith = sinon.stub();
+    it('should override authentication route', async function () {
+      // when
+      await route.beforeModel();
 
+      // then
+      expect(route.authenticationRoute).to.equal('inscription');
+    });
+
+    it('should call parent’s beforeModel and transition to authenticationRoute', async function () {
       // when
-      await route.beforeModel(transition);
+      await route.beforeModel();
 
       // then
-      expect(route.state.hasUserSeenJoinPage).to.be.undefined;
+      sinon.assert.calledWith(route.router.transitionTo, 'inscription');
+    });
+
+    context('when campaign belongs to pole emploi and user is not connected with pole emploi', function () {
+      it('should override authentication route with login-pe', async function () {
+        // given
+        route.session.data.externalUser = 'some external user';
+        campaign.organizationIsPoleEmploi = true;
+
+        // when
+        await route.beforeModel();
+
+        // then
+        expect(route.authenticationRoute).to.equal('login-pe');
+      });
+    });
+
+    context('when campaign is SCO restricted and user is neither authenticated nor external', function () {
```

Comment:
`when campaign is SCO restricted and user is neither authenticated from Pix nor a user from an external platform`